src/sentry/sentry_apps/api/endpoints/sentry_app_requests_v2.py

```diff
@@ -0,0 +1,178 @@
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    permission_classes = (SentryAppStatsPermission,)
+
+    def get(self, request: Request, sentry_app: SentryApp) -> Response:
+        """
+        :qparam string eventType: Optionally specify a specific event type to filter requests
+        :qparam bool errorsOnly: If this is true, only return error/warning requests (300-599)
+        :qparam string organizationSlug: Optionally specify an org slug to filter requests
+        :qparam string start: Optionally specify a date to begin at. Format must be YYYY-MM-DD HH:MM:SS
+        :qparam string end: Optionally specify a date to end at. Format must be YYYY-MM-DD HH:MM:SS
+        """
+        serializer = IncomingRequestSerializer(data=request.GET)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        serialized = serializer.validated_data
+
+        event_type = serialized.get("eventType")
+        errors_only = serialized.get("errorsOnly")
+        org_slug = serialized.get("organizationSlug")
+        start_time = serialized.get("start")
+        end_time = serialized.get("end")
+
+        organization = None
+        if org_slug:
+            try:
+                organization = OrganizationMapping.objects.get(slug=org_slug)
+            except OrganizationMapping.DoesNotExist:
+                return Response({"detail": "Invalid organization."}, status=400)
+
+        requests: list[RpcSentryAppRequest] = []
+        region_filter: SentryAppRequestFilterArgs = {}
+        control_buffer = SentryAppWebhookRequestsBuffer(sentry_app)
+        control_errors_only = region_filter["errors_only"] = errors_only
+
+        def filter_and_append_requests(unfiltered_requests: list[RpcSentryAppRequest]) -> None:
+            for i, req in enumerate(unfiltered_requests):
+                if filter_by_date(req, start_time, end_time) and filter_by_organization(
+                    req, organization
+                ):
+                    requests.append(BufferedRequest(id=i, data=req))
+
+        def get_buffer_requests_for_control(event_type) -> list[RpcSentryAppRequest]:
+            control_requests = [
+                serialize_rpc_sentry_app_request(req)
+                for req in control_buffer.get_requests(
+                    event=event_type, errors_only=control_errors_only
+                )
+            ]
+            filter_and_append_requests(control_requests)
+
+        def get_buffer_requests_for_regions() -> list[RpcSentryAppRequest]:
+            for region_name in find_all_region_names():
+                buffer_requests = app_request_service.get_buffer_requests_for_region(
+                    sentry_app_id=sentry_app.id,
+                    region_name=region_name,
+                    filter=region_filter,
+                )
+                if buffer_requests is not None:
+                    filter_and_append_requests(buffer_requests)
```

Comment:
Instead of nesting these closures, could these live in a separate class/service that handles retrieving buffer requests? I could see a case where we use these outside of this endpoint.